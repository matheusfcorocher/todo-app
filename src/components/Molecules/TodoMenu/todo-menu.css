@import '../../../foundations/colors/colors.css';

body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.todo-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}

.list-icon-container {
  width: 100px;
  height: 100px;
}

.list-icon {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  fill: var(--white-color);
  stroke: var(--white-color) !important;
  stroke-width: 0.1;
  background: transparent;
}

.checklist-icon {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  fill: var(--dark-color);
  stroke: var(--dark-color) !important;
  stroke-width: 0.1;
  background: transparent;
}

.complete-button {
  width: 3rem;
  height: 3rem;
  padding: 0.2rem;
  border-radius: 0.2rem;
  display: flex;
  align-items: center;
  margin-right: 1rem;
  background-color: var(--primary-color);
}

.title-input {
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  border-radius: 0.1rem;
  border: none;
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 1rem;
  font-weight: 400;
}

.title-input[type="text"] {
  padding: 0.2rem;
}

.title-input::placeholder {
  padding: 0.2rem;
}