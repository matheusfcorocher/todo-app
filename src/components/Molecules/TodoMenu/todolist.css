body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.todo-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}

.complete-button {
  margin-right: 1rem;
}

.title-input {
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  padding: 0;
  border: none;
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 1rem;
  font-weight: 400;
}