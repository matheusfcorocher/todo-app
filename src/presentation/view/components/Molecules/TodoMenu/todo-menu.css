@import '../../../styles/colors/colors.css';
@import '../../../styles/round-corners/round-corners.css';
@import '../../../styles/spacing/spacing.css';
@import '../../../styles/typography/typography.css';

.todo-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}

.list-icon-container {
  width: 100px;
  height: 100px;
}

.list-icon {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  fill: var(--dark-color);
  background: transparent;
}

.checklist-icon {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  fill: var(--gray-color);
  background: transparent;
}

.complete-button {
  width: 2rem;
  height: 2rem;
  padding: var(--spacing-xs);
  border-radius: var(--spacing-xs);
  display: flex;
  align-items: center;
  /* margin-right: var(--spacing-sm); */
  background-color: transparent;
}

.title-input {
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  border-radius: var(--round-corner-sm);
  border: none;
  font-family: var(--ff-serif);
  font-size: var(--fs-300);
}

.title-input[type="text"] {
  padding: var(--spacing-xs);
}

.title-input::placeholder {
  padding: var(--spacing-xs);
}