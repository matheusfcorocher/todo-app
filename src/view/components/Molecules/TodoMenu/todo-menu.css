@import '../../../foundations/colors/colors.css';
@import '../../../foundations/round-corners/round-corners.css';
@import '../../../foundations/spacing/spacing.css';
@import '../../../foundations/typography/typography.css';

.todo-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}

.list-icon-container {
  width: 100px;
  height: 100px;
}

.list-icon {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  fill: var(--dark-color);
  stroke: var(--dark-color) !important;
  stroke-width: 0.1;
  background: transparent;
}

.checklist-icon {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  fill: var(--dark-color);
  stroke: var(--dark-color) !important;
  stroke-width: 0.1;
  background: transparent;
}

.complete-button {
  width: 3rem;
  height: 3rem;
  padding: var(--spacing-xs);
  border-radius: var(--spacing-xs);
  display: flex;
  align-items: center;
  margin-right: var(--spacing-md);
  background-color: var(--primary-color);
}

.title-input {
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  border-radius: var(--round-corner-sm);
  border: none;
  font-family: var(--ff-serif);
  font-size: var(--fs-300);
}

.title-input[type="text"] {
  padding: var(--spacing-xs);
}

.title-input::placeholder {
  padding: var(--spacing-xs);
}