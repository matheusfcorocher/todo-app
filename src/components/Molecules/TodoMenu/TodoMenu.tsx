import React, { useState } from 'react';
import { createImmediatelyInvokedFunctionExpression } from 'typescript';
import { handleCreateTodo } from '../../../App';
import '../../../App.css';
import IconButton from '../../Atoms/IconButton/IconButton';
import CheckListIcon from '../../Atoms/icons/CheckListIcon/CheckListIcon';
import ListIcon from '../../Atoms/icons/ListIcon/ListIcon';
import './todo-menu.css';

interface TodoMenuProps {
    handleCreateTodo: handleCreateTodo;
    isAllTodosCompleted: boolean;
}

function TodoMenu({ handleCreateTodo, isAllTodosCompleted }: TodoMenuProps) {
    const [title, setTitle] = useState<string>("");

    function handlePress(event : React.KeyboardEvent<HTMLInputElement>) {
        if(event.key == "Enter") {
            handleCreateTodo(title);
            setTitle("");
        }
    }
    
    return (
        <div role="group" className={`todo-menu`}>
            {
                isAllTodosCompleted ?
                    (<IconButton className="complete-button" handleFunction={function (): void {
                        throw new Error('Function not implemented.');
                    }}>
                        <ListIcon className="list-icon" />
                    </IconButton>)
                    :
                    (<IconButton className="complete-button" handleFunction={function (): void {
                        throw new Error('Function not implemented.');
                    }}>
                        <CheckListIcon className="checklist-icon" />
                    </IconButton>)
            }
            <input
                type="text"
                role="textbox"
                placeholder='What needs to be done?'
                value={title}
                name="title"
                className={`title-input`}
                onChange={(e) => setTitle(e.target.value)}
                onKeyDown={(event) => handlePress(event)}
            />
        </div>
    );
}

export default TodoMenu;
