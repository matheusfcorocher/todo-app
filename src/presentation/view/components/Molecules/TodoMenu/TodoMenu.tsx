import React, { useState } from "react";
import { animated, useTransition } from "@react-spring/web";
import { TodoTaskControllerReturnType } from "../../../../controllers/TodoTaskController";
import IconButton from "../../Atoms/IconButton/IconButton";
import ArrowDownIcon from "../../Atoms/icons/ArrowDownIcon/ArrowDownIcon";
import "./todo-menu.css";

interface TodoMenuProps {
  todoTaskController: TodoTaskControllerReturnType;
}

function TodoMenu({ todoTaskController }: TodoMenuProps) {
  const [title, setTitle] = useState<string>("");
  const {
    handleAddTodoTask,
    handleIncompleteAllTodoTasks,
    handleCompleteAllTodoTasks,
    getIsAllTodoTaskCompleted,
    getIsTodoTasksNotEmpty,
  } = todoTaskController;
  function handlePress(event: React.KeyboardEvent<HTMLInputElement>) {
    if (event.key == "Enter" && title !== "") {
      handleAddTodoTask(title);
      setTitle("");
    }
  }

  //Animations

  function convertRemToPixels(rem: number) {    
    return rem * parseFloat(getComputedStyle(document.documentElement).fontSize);
  }

  //Icon Button Animation
  const iconButtonTransition = useTransition(getIsTodoTasksNotEmpty(), {
    from: { opacity: 0, width: 0 },
    enter: { opacity: 1, width: convertRemToPixels(2)},
    leave: { opacity: 0, width: 0 },
  });

  return (
    <div role="group" className={`todo-menu`}>
      {iconButtonTransition((style, item) =>
        item ? (
          <animated.div style={style}>
            {getIsAllTodoTaskCompleted() ? (
              <IconButton
                title="Uncheck all todos"
                className="complete-button"
                handleFunction={() => handleIncompleteAllTodoTasks()}
              >
                <ArrowDownIcon className="list-icon" />
              </IconButton>
            ) : (
              <IconButton
                title="Check all todos"
                className="complete-button"
                handleFunction={() => handleCompleteAllTodoTasks()}
              >
                <ArrowDownIcon className="checklist-icon" />
              </IconButton>
            )}
          </animated.div>
        ) : (
          ""
        )
      )}
      <input
        type="text"
        role="textbox"
        placeholder="What needs to be done?"
        value={title}
        name="title"
        className={`title-input ${getIsTodoTasksNotEmpty() ? "hidden" : ""}`}
        onChange={(e) => setTitle(e.target.value)}
        onKeyDown={(event) => handlePress(event)}
      />
    </div>
  );
}

export default TodoMenu;
