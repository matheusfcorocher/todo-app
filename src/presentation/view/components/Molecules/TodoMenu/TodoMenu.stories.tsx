import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import TodoMenu from './TodoMenu';

export default {
  title: 'Molecules/TodoMenu',
  component: TodoMenu,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0 }}>
        <Story />
      </ul>
    ),
  ]
} as ComponentMeta<typeof TodoMenu>;

const Template: ComponentStory<typeof TodoMenu> = (args) => <TodoMenu {...args} />;

export const Default = Template.bind({});
Default.args = {
  isThereAnyTodoTaskCompleted: false
};

export const IncompleteAllTodos = Template.bind({});
Default.args = {
  isThereAnyTodoTaskCompleted: true
};
