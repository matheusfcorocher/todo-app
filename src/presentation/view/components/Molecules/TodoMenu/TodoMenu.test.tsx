import React from 'react';
import { render, screen } from '@testing-library/react';
import TodoMenu from './TodoMenu';

describe("::Components ::Molecules ::TodoMenu", () => {
  test('renders TodoMenu', () => {
      render(<TodoMenu todoTaskController={{
        handleAddTodoTask: function (title: string): void {
          throw new Error('Function not implemented.');
        },
        handleDeleteTodoTask: function (id: string): void {
          throw new Error('Function not implemented.');
        },
        handleUpdateTodoTaskTitle: function (id: string, newTitle: string): void {
          throw new Error('Function not implemented.');
        },
        handleUpdateTodoTaskState: function (id: string, state: boolean): void {
          throw new Error('Function not implemented.');
        },
        handleCompleteAllTodoTasks: function (): void {
          throw new Error('Function not implemented.');
        },
        handleIncompleteAllTodoTasks: function (): void {
          throw new Error('Function not implemented.');
        },
        handleDeleteAllCompletedTodoTasks: function (): void {
          throw new Error('Function not implemented.');
        }
      }} isThereAnyTodoTaskCompleted={false}/>);
      const todoMenu = screen.getByRole("group");
      expect(todoMenu).toMatchSnapshot();
  })
})