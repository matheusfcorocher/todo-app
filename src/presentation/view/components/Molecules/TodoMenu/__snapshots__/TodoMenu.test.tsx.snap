// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`::Components ::Molecules ::TodoMenu renders TodoMenu 1`] = `
<div
  class="todo-menu"
  role="group"
>
  <button
    class="icon-button complete-button"
    role="button"
    title="Uncheck all todos"
  >
    <svg
      class="list-icon"
      preserveAspectRatio="none"
      role="image"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        fill=""
      />
    </svg>
  </button>
  <input
    class="title-input"
    name="title"
    placeholder="What needs to be done?"
    role="textbox"
    type="text"
    value=""
  />
</div>
`;
